core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -354,10 +354,10 @@ public static class AuthSession {
     private static int tokenRefreshNumRetries = 5;
```

Comment:
This could be volatile also, maybe make `refresh()` synchronized as well?

Reply:
This is for testing only, so I think it's okay.